### PR TITLE
lint was updated per linked issues. This fixes new requirement

### PR DESCRIPTION
**Pull request by endophage**
https://github.com/golang/lint/pull/245
https://github.com/golang/go/issues/17293

Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

**Comment by endophage** [comment]
@cyli @riyazdf @HuKeping not sure why codecov failed but the diff has 100% coverage and coverage wouldn't have changed for this change anyway. If we can get this merged ASAP other PRs will be able to pass linting (once rebased)

File: notary.go [hunk 1]
```diff
@@ -5,3 +5,8 @@ package notary
 // confirmation), createNew will be true. Attempts is passed in so that implementers
 // decide how many chances to give to a human, for example.
 type PassRetriever func(keyName, alias string, createNew bool, attempts int) (passphrase string, giveup bool, err error)
+
+// CtxKey is a wrapper type for use in context.WithValue() to satisy golint
+// https://github.com/golang/go/issues/17293
+// https://github.com/golang/lint/pull/245
+type CtxKey string
```
Comment (cyli) [inline notary.go]: Non-blocking:  wondering if we can make constants for the 4 CTX keys we use:  `cryptoService`, `metaStore`, `repo` (maybe we should change this to "GUN" though), and `keyAlgorithm`.  

Even better is if we can unexport the type and just export the constants, but lint might complain.

At any rate, happy to have this done in another PR.

Comment (endophage) [inline notary.go]: ooooh, I do like that idea! Lets unblock circle for now and I'll submit that tomorrow.

File: notary.go [hunk 2]
```diff
@@ -5,3 +5,8 @@ package notary
 // confirmation), createNew will be true. Attempts is passed in so that implementers
 // decide how many chances to give to a human, for example.
 type PassRetriever func(keyName, alias string, createNew bool, attempts int) (passphrase string, giveup bool, err error)
+
+// CtxKey is a wrapper type for use in context.WithValue() to satisy golint
```
Comment (cyli) [inline notary.go]: Oops, sorry one last non-blocking nitpick:  "satisfy"

Comment (endophage) [inline notary.go]: Huh, wonder why misspell didn't catch that. Fixing

File: notary.go (hunk 1 above)
Comment (endophage) [inline notary.go]: https://github.com/docker/notary/issues/989
